resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilder.scala

```diff
@@ -51,7 +57,13 @@ private[spark] class KubernetesDriverBuilder(
     provideJavaStep: (
       KubernetesConf[KubernetesDriverSpecificConf]
         => JavaDriverFeatureStep) =
-    new JavaDriverFeatureStep(_)) {
+    new JavaDriverFeatureStep(_),
+    provideTemplateVolumeStep: (KubernetesConf[_ <: KubernetesRoleSpecificConf]
+      => TemplateVolumeStep) =
+    new TemplateVolumeStep(_),
+    provideInitialSpec: KubernetesConf[KubernetesDriverSpecificConf]
```

Comment:
Why not `provideInitialPod` to be consistent with the executor builder?

Reply:
I think it's because we need the `KubernetesDriverSpec` object instead of just the pod? which includes things like the entire sparkConf map

Reply:
You can make the initial pod and wrap it in the `KubernetesDriverSpec` object.